src/psyclone/tests/psyir/frontend/fparser2_data_statement_test.py

```diff
@@ -0,0 +1,128 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+'''Tests Fortran data statements in the fparser2 PSyIR front-end. Data
+statements are represented as an UnsupportedFortranType. The tests
+here cover a few variations of the data statement, in case that at
+some stage we need more information.
+'''
+
+import pytest
+from fparser.common.readfortran import FortranStringReader
+from fparser.two.Fortran2003 import Specification_Part
+from psyclone.psyir.frontend.fparser2 import Fparser2Reader
+from psyclone.psyir.nodes import Routine
+from psyclone.psyir.symbols import UnsupportedFortranType
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_simple_data_statement():
+    ''' Test that data statements are correctly captured. '''
+
+    # Create a dummy test routine
+    routine = Routine("test_routine")
+    symtab = routine.symbol_table
+    processor = Fparser2Reader()
+
+    # Test with a single namelist
+    reader = FortranStringReader('''
+        data a /1/''')
+    fparser2spec = Specification_Part(reader)
+    processor.process_declarations(routine, fparser2spec.content, [])
+    # There is a data_stmt in the symbol table:
+    data = symtab.lookup("_PSYCLONE_INTERNAL_DATA_STMT")
+    assert isinstance(data.datatype, UnsupportedFortranType)
+    assert data.datatype.declaration == "DATA a / 1 /"
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_multiple_data_statement():
+    ''' Test that data statements are correctly captured. '''
+
+    # Create a dummy test routine
+    routine = Routine("test_routine")
+    symtab = routine.symbol_table
+    processor = Fparser2Reader()
+
+    # Test with a single namelist
```

Comment:
s/namelist/data statement/ :-)

Reply:
...with multiple initialisations.

Reply:
Done and done ... this gives the totally wrong impression I copied that test from somewhere :)